sbt-dumpschema/src/main/scala/org/virtuslab/akkasaferserializer/DumpSchemaPlugin.scala

```diff
@@ -0,0 +1,30 @@
+package org.virtuslab.akkasaferserializer
+
+import sbt.Keys._
+import sbt.{Def, _}
+
+object DumpSchemaPlugin extends AutoPlugin {
+
+  object autoImport extends DumpSchemaKeys {
+    val baseDumpSchemaSettings: Seq[Def.Setting[_]] = DumpSchemaPlugin.baseDumpSchemaSettings
+  }
+  import autoImport.{baseDumpSchemaSettings => _, _}
+
+  override def projectSettings: Seq[Def.Setting[_]] = additionalSettings ++ dumpSchemaSettings
+
+  override def globalSettings: Seq[Def.Setting[_]] = Nil
+
+  lazy val additionalSettings = Seq(
+    libraryDependencies += compilerPlugin((dumpSchema / dumpSchemaPlugin).value),
+    scalacOptions ++= Seq(s"-P:dump-schema-plugin:${target.value / "dump"}"))
+
+  lazy val dumpSchemaSettings: Seq[Def.Setting[_]] = baseDumpSchemaSettings
+
+  lazy val baseDumpSchemaSettings: Seq[Def.Setting[_]] = Seq(
+    DumpSchema.assemblyTask(dumpSchema),
+    dumpSchema / dumpSchemaOutputPath := { target.value / (dumpSchema / dumpSchemaFilename).value },
```

Comment:
Hmm could you attach here a sample output file, preferably constructed from a couple of different Scala source files?

Reply:
Output will be an array of objects like:
```json
{
  "isTrait":true,
  "name":"TraitWithAnnotation",
  "annotation":[
    {
      "name":"org.random.project.TestAnn(1, 2)"
    },
    {
      "name":"org.random.project.TestAnn(1, project.this.TestAnn.<init>$default$2)"
    }
  ],
  "fields":[
    {
      "name":"c",
      "typeName":"AdditionalData"
    },
    {
      "name":"b",
      "typeName":"String"
    },
    {
      "name":"a",
      "typeName":"Long"
    }
  ],
  "parents":[
    "Object",
    "Data"
  ]
}
```

Reply:
I'd make sure the output is _canonicalized_, esp. that classes are sorted alphabetically (not sure if we should sort fields within classes: probably not!)... actually, what does the current ordering of classes within the output depend on? 🤔